### PR TITLE
[RM Ch03] Storage resource description

### LINKED ISSUE #1851: [RM Ch03] Storage resource description 3.2.3
Section3.2.3 Storage
The description of storage is too restrictive, addressing only block device. The description does not reflect the diversity of storage

### PR DESCRIPTION
**Pull request by karinesevilla**
Fixes #1851 
Section 3.2.3 Storage
The original storage resource description addresses only about Block device storage.
The intent of this proposal is to make it more generic and to reflect storage diversity

### CODE REVIEW

**Comment by karinesevilla** [comment]
@walterkozlowski The first intent was to correct the previous description which focused only on Block device and persistent storage.
You are right, we'll have to elaborate on storage, but in my opinion it must be rather done within chapters 4 and 5

**Comment by wmk-admin** [comment]
@pgoyal01 @ulikleber @ASawwaf please review and approve if you can

**Comment by walterkozlowski** [comment]
@pgoyal01 @TFredberg please review and approve you can, 
I agree with @karinesevilla that we need to elaborate on storage in Chapter 4., @petorre please assist @karinesevilla in this

File: doc/ref_model/chapters/chapter03.md
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter03.md:102]: availability is typically in terms of percentage

```suggestion
A workload can request different types of storage based on data longevity: persistent or ephemeral storage.
```

File: doc/ref_model/chapters/chapter03.md
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter03.md:105]:
```suggestion
There are multiple storage performance requirements such as latency, IOPS and capacity. For example, a workload may require one of its storage device to provide low latency, high IOPS and very large/hige storage capacity (terabytes of data).
```

File: doc/ref_model/chapters/chapter03.md
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter03.md:107]:
```suggestion
High IOPS oriented storage is for workloads requiring lots of read/write actions.
```

File: doc/ref_model/chapters/chapter03.md
Comment (pgoyal01) [reply doc/ref_model/chapters/chapter03.md:105]: May I suggest that capacity be separated from performance -- see storage comment below

File: doc/ref_model/chapters/chapter03.md
Comment (karinesevilla) [reply doc/ref_model/chapters/chapter03.md]: @pgoyal01 Thanks, adopted!

File: doc/ref_model/chapters/chapter03.md
Comment (karinesevilla) [reply doc/ref_model/chapters/chapter03.md:105]: @pgoyal01 Thanks, suggestion adopted
I will modify the typo hige->huge

File: doc/ref_model/chapters/chapter03.md
Comment (karinesevilla) [reply doc/ref_model/chapters/chapter03.md:107]: @pgoyal01 Thanks, adopted!

File: doc/ref_model/chapters/chapter03.md
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter03.md:116]:
```suggestion
| `performance`       | latency, IOPS, capacity                            |
```

File: doc/ref_model/chapters/chapter03.md
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter03.md:119]:
```suggestion
| `size`              | size in GB                                                       |
```

File: doc/ref_model/chapters/chapter03.md
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter03.md:110]: do we also need a device type (HDD, SSD, ...)?
do we need to include SDS (e.g., ceph)?

File: doc/ref_model/chapters/chapter03.md
Comment (karinesevilla) [reply doc/ref_model/chapters/chapter03.md:119]: @pgoyal01 Adopted

File: doc/ref_model/chapters/chapter03.md
Comment (karinesevilla) [reply doc/ref_model/chapters/chapter03.md:116]: @pgoyal01 Adopted

File: doc/ref_model/chapters/chapter03.md
Comment (karinesevilla) [reply doc/ref_model/chapters/chapter03.md:110]: @pgoyal01 I think the chapter 3 must keep only the modelling aspect without going into technologies. For me, it is not the good place to describe the device types and SDS. 
But you are right these topics must be addressed.  Can it be integrated into chapter 4, adding a section under 4.1.4 internal Infra capabilities?

Comment (walterkozlowski) [reply doc/ref_model/chapters/chapter03.md:110]: I agree with @karinesevilla that such details would be better to have in Chapter 4.  @pgoyal01, if you agree, I would ask @karinesevilla to raise an issue to create a subsection 0f 4.1.4 to deal with the technology aspects of storage

Comment (pgoyal01) [reply doc/ref_model/chapters/chapter03.md:110]: @walterkozlowski @karinesevilla I agree with the approach to raise an issue and insert this in Ch4. Thanks

**Comment by petorre** [comment]
> I agree with @karinesevilla that we need to elaborate on storage in Chapter 4., @petorre please assist @karinesevilla in this

@walterkozlowski We started and have snapshot of where "storage" is mentioned throughout documents. That we should review on one of RM calls and agree how to proceed before raising needed PRs.

**Review by petorre — APPROVED** [review]

**Pull request merged**
